### ISSUE TITLE
failure to launch

### ISSUE THREAD

**Issue by FatherfoxStrongpaw** [myshell-ai/AIlice#65]
failure to launch. your requirements.txt has syntax errors and the build wheel fails, particularly with cmake. i just spent more than $20 with claude through cline trying to fix it and no joy.  in the last 6 months i've only had your project up and running 4 times and that after getting lucky. have yet to get scripter running without loading a new DB and rarely get to a web page from docker.

**Comment by stevenlu137** [comment]
Please provide the details of your runtime environment so that I can reproduce the issue. If AIlice installation or startup fails, the error message is also necessary for troubleshooting. As far as I know, the main issue with installing AIlice on Ubuntu is that llama-cpp-python occasionally causes installation failures. I am currently looking for a more stable alternative to address this problem. Your situation seems a bit different, but I don't have enough information to reproduce your issue.

**Comment by stevenlu137** [comment]
There is indeed a syntax error in the requirements.txt, which may have been caused by the PR submitter using AI to generate the code. For some reason, I didn’t carefully review this change, which led to the faulty code being merged into the repository. It is more convenient to maintain a single approach, so I have deleted this file. Please use pip install -e . to install AIlice.

**Comment by stevenlu137** [comment]
Hi, thank you for your suggestions. 
I'm also considering improving the UI design, but it might be different from the design approach of similar products. AIlice will focus on simplicity and universality, so it won't include too many complex options. Instead, all operations will be encapsulated within rich-text conversations. 
Providing a better Docker installation method will be a very useful option for many users, and I will work on it. However, it’s important to note that for experienced users, running locally can provide a better experience, especially when you frequently need AIlice to write code or help configure systems. The installation failure issue might still be largely related to llama-cpp-python, and I need to find a better alternative to run the embedding models. 

AIlice will not be abandoned; the occasional stagnation last year was related to some personal misfortunes, but perhaps the worst days are behind us, and the project will be updated again.

**Comment by stevenlu137** [comment]
@FatherfoxStrongpaw  The latest code already supports running Ailice inside a Docker container completely. Although the BROWSE function still has some issues, the installation process should be much simpler now. You can give it a try.

**Comment by FatherfoxStrongpaw** [comment]
i just tried to delete my last ailice directory and i can't because of a null file, so i renamed it and pulled a fresh git clone. i was happily supprised when i got only a couple errors on the pip install, but it allowed the ailice_web to fire up ailice and ask for an openrouter api key. i gave it and ailce started right up and was working fantastic... until i ran out of tokens. 

when i added more, it crapped out on me. my copy/paste  glitched so i didn't get the error. after restarting, deleting ailice dir and steven lu directory and another clone/install, no mater what i do other than give it  an openai key all i get is asked for an openai key only and then:


C:\Users\Admin\AIlice>ailice_web --modelID=openrouter:qwen/qwen-2-72b-instruct
********************** Initialize *****************************
config.json is located at C:\Users\Admin\AppData\Local\Steven Lu\ailice
********************** End of Initialization *****************************
In order to simplify installation and usage, we have set local execution as the default behavior, which means AI has complete control over the local environment. To prevent irreversible losses due to potential AI errors, you may consider one of the following two methods: the first one, run AIlice in a virtual machine; the second one, install Docker, use the provided Dockerfile to build an image and container, and modify the relevant configurations in config.json. For detailed instructions, please refer to the documentation.
If you find that ailice is running slowly or experiencing high CPU usage, please run `ailice_turbo` to install GPU acceleration support.
killing proc with PID 6548
killing proc with PID 12672
killing proc with PID 14344
killing proc with PID 16980
killing proc with PID 17128
killing proc with PID 17392
killing proc with PID 17648
storage  started.
browser  started.
arxiv  started.
google  started.
duckduckgo  started.
scripter  started.
computer  started.
 * Serving Flask app 'ailice.ui.app'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
Press CTRL+C to quit
127.0.0.1 - - [22/Jan/2025 15:37:15] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [22/Jan/2025 15:37:16] "GET /list_histories HTTP/1.1" 200 -
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
We now start the vector database. Note that this may include downloading the model weights, so it may take some time.
Vector database has been started. returned msg: vector database under C:\Users\Admin\AppData\Local\Steven Lu\ailice\ailice_1737589041\storage is opened. model: nomic-ai/nomic-embed-text-v1-GGUF, gguf: nomic-embed-text-v1.Q8_0.gguf
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
127.0.0.1 - - [22/Jan/2025 15:37:21] "GET /new_chat HTTP/1.1" 200 -
127.0.0.1 - - [22/Jan/2025 15:37:22] "POST /chat HTTP/1.1" 200 -
127.0.0.1 - - [22/Jan/2025 15:37:23] "GET /static/User.JPG HTTP/1.1" 304 -
ASSISTANT_AIlice:  127.0.0.1 - - [22/Jan/2025 15:37:23] "GET /static/AIlice.png HTTP/1.1" 304 -

Exception in thread Thread-5:
Traceback (most recent call last):
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\threading.py", line 1052, in _bootstrap_inner
    self.run()
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\threading.py", line 989, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Admin\AIlice\ailice\core\AProcessor.py", line 123, in __call__
    ret = self.llm.Generate(prompt, proc=loggerMsg, endchecker=self.interpreter.EndChecker, temperature = config.temperature)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Admin\AIlice\ailice\core\llm\AModelChatGPT.py", line 28, in Generate
    for chunk in self.client.chat.completions.create(model=self.modelName,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\site-packages\openai\_utils\_utils.py", line 279, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\site-packages\openai\resources\chat\completions.py", line 859, in create
    return self._post(
           ^^^^^^^^^^^
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\site-packages\openai\_base_client.py", line 1283, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\site-packages\openai\_base_client.py", line 960, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "C:\Users\Admin\AppData\Local\Programs\Python\Python312\Lib\site-packages\openai\_base_client.py", line 1064, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.AuthenticationError: Error code: 401 - {'error': {'message': 'No auth credentials found', 'code': 401}}

gods ailice is sooo close to perfect! F*CK I WISH I WAS A CODE MONKEY! 
and trust me, i feel ya with life getting in the way. it still is for me, but being on a fixed income isn't helping. i'm stuck using free and open source tools as i stumble around in the dark, but i'm working on it. one thing i'd really like you to copy from zero is being able to get ailice started and having a settings button to plug in your options for api keys, models/etc as well as the file configurations for advanced users. 

of all the frameworks i've tried in the last 8 months AiLiceAI and Agent_Zero are hands down the 2 best at their respective use cases and target audiences. if my last name was Musk, you'd be fully funded today! i've been following AiLiceAI since 2004 or 06. AiLiceAI, UltraHAL and CLEO were my inspiration to switch from robotics to AI (My personal Project D.I.A.N.A. Dynamic Intuitive Autonomous Neural Architecture). i had to go on hiatus in 2012 to take care of my wife who passed in 2016 and i'm just now getting back in the game  last april. LLM's are the game changer and your ailice is almost everything i was building back then. 

Never give up, never surrender!  Your almost there! And I'm not giving up on you!  

Lee C. (AKA: Fatherfox)

P.S. thank god your not trying to pull this off in LISP and PHP! *^_^*

**Comment by stevenlu137** [comment]
I think you must have entered the wrong API key, for example, problems occur when using an OpenAI API key for OpenRouter models. The solution is to directly open the configuration file(C:\Users\Admin\AppData\Local\Steven Lu\ailice\config.json), search for the model name, and modify the API key configuration. You're right, we need to provide API key settings in the UI, as the current approach makes it too easy for users to make mistakes. I will add these new features to the UI in the future.

Today I just upgraded AIlice's voice dialogue functionality, which now has more practical value (though it requires a more powerful GPU), you might be interested in that. Thank you for your encouragement, and your story also makes me sad. In 2024, I lost three family members, including my father, and currently my only motivation for updating AIlice is to advance AI technology and take revenge against myocarditis and autoimmune diseases (AIlice's biggest application case so far has been helping me investigate my father's illness - I spent thousands of dollars in token fees, and although we couldn't save him, it gave me a better grasp of AIlice's capabilities). I will continue to push this project to solve more complex problems, optimize the user interface, and expand the user base, especially academic users.

I noticed you mentioned following the AiLiceAI project since 200x, but I can't find any information about this project on the internet (AIlice was built by me from scratch in 2023, and I don't know about AiLiceAI). Could you tell me more about this project?

**Comment by FatherfoxStrongpaw** [comment]
after my wife died i gave both my inlaws end of life care. my mother-n-law passed in 2022 and my father-n-law in 23. i can relate. my wife had MS and was diagnosed in 2001. by the end of 2002 she was bound to a wheelchair. i dropped out of security and robotics to concentrate on networking and tech support to stay home with her and got into AI as a hobby. i thought your project was this:
https://www.chatbots.org/chatbot/a.l.i.c.e/ .

my bad, but all in all, i like your vision better. 

i'm glad your sticking with it. i at least got it to "kinda" work. "one must look to the good in any situation" or so i'm told. i'm working on some modules for it. mainly loop detection, error handling, versioning and a few other things. just fragged my VS code getting rid of pythagora. it's not ready for prime time either, so i'm going back to cline/deepseek r1 and agent-zero/chatgpt 4o. 

i was <this---> close to getting a functional merge of ailice and zero with elements from my original assistant from back in the day (Project D.I.A.N.A. Dynamic Autonomous Intuitive Neural Architecture) i started on a diy beowolf cluster until cline/claude wiped out all my functional code and even found all the debugging logs, but left all the prompts and non-executable code and empty directories behind. it fragged all the files before deleting making sure when i did a forensic file recovery, there was nothing useable to recover. that was no oops. 

anyhu, here's some output from todays challenge in an attachment. i call it "where's the breaks on this thing?" but hey, it's actually working thanks to your hard work. ailice may be pissing me off, but then, a lot of it is the "guard rails" in the llm's. aka: "nevermind the gun, beware the owner!" the llm's are only doing what the creators trained them to do. 

your project may be the least developed of all the frameworks i'm playing with, but like i said, it has the most potential and i say that with 20 years of authority. (i sucked at coding back then too. >;) )

[where's the breaks on this thing.txt](https://github.com/user-attachments/files/18554108/where.s.the.breaks.on.this.thing.txt)

so what i'm mainly trying to incorporate now is error detection, self healing, loop detection/prevention/intervention, recursive self improvement and llm independent behavior. dragsters are fun at top speed, but when the breaks don't work, that's the wrong time to find out your buddy packed his dirty laundry in your shoot bag as a practical joke. claude is definitely NOT my friend.

**Comment by stevenlu137** [comment]
I’m sorry to hear about your experiences. I spent three months in the hospital and met all kinds of people suffering, and I think AI might be dangerous in the future, but that’s still uncertain. What’s certain is that it can help us alleviate the suffering we face right now. By the way, have you ever shared your D.I.A.N.A. project publicly? If so, I’d like to take a look.

As for the problem you encountered today, it’s something I’ve always been aware of, but I haven’t yet made a decision on how to address it. The reason agents ignore your task termination prompt is that only one agent sees your prompt: the one currently working at the time you issue it. When it follows the instruction to end its work, the control flow returns to the parent agent, which, due to a lack of context, mistakenly assumes the task wasn't completed correctly and prompts the agent to continue working. This issue isn't limited to just the task termination operation—any user intervention prompt tends to cause issues to some extent, because the child agent receives more information from the user, while the parent agent lacks this information. When the user changes their requirements in the interrupt prompt, conflicts arise.

How to solve this problem? What we can be sure of is that interrupt intervention is a temporary measure that will eventually disappear. In the future, we might face agent execution flows that exceed human reading speed, along with a large number of agents running in parallel—something we can’t handle. So, the measures we introduce now must be temporary and shouldn’t create lasting historical baggage.

A possible solution is to add a prompt that allows the child agent to provide the relevant information about the user’s interrupt intervention back to the parent agent, ensuring that the information is fully shared. This is the most suitable approach I can think of right now, as prompts are more flexible than code and are better suited for making priority changes. If there are no issues, I will modify the code to implement it.

But this doesn’t completely solve the problem, as agents might not follow the prompt. So, a method to forcefully end the task without using Ctrl+C makes sense, and we will consider adding this feature as well.

**Comment by FatherfoxStrongpaw** [comment]
my latest 7 iterations of Diana all got fragged or deleted from bad agent executions by me (be careful what you wish for). cline with claude killed the last version like i said. after 8 months of trial and error, i finally had the closest cleanest functioning framework by stopping trying to reinvent the wheel and starting with your ailce and cherry picking the best code and functions from my favorite 10 frameworks all cloned into a master directory for context, mega XL'ed ailice, and was working on getting ailice's start script working with a work around the cmake failure (problem on my end, now fixed). cline decided to change all the files i finished debugging and started changing file permissions, that's when i new something was wrong. it "cleaned up" the repository, but was selective. it only hit the files that had executable code, but left everything else, text files, prompts and the like. 

now that your getting the install and start scripts stable, i'm actually able to really see what ailice can do. Ailice is the closest to what i originally wanted in diana as a core logic. and zero has the best framework/ui. together with the error handling and a few tweaks, i think we can get something worthy of your father and my late wife. My intention to bring back Daina was to have a digital assistant to help Vetrans and the disabled with a sort of "Her/Sonny/Andrew" kind of digital companion that was multi-agent (exactly like Ailce) with a clean ui and stable framework (like zero).  

as for this issue, i've had several clean reinstalls, so this issue can close at your leisure unless there's anything else you'd like me to look into. also, if your interested in further discussions on adding some guardrails or a possible colab, i'm at fatherfox@yahoo.com. I intend to keep working on ailice standalone as well as trying her out as a fork and using her core in a couple variants. 

your the only one i think that's on the right track for the most robust virtual assistant and we share similar goals. the memory of your father and my wife and mutual burnout. add to that, your the first real person to actually inspire me for something besides boredom. Stay awesome and Ailice will too! you got a real world beta tester if nothing else.

**Comment by stevenlu137** [comment]
Current AI tools, including Ailice, have not yet reached a level of reliability where they can be fully trusted. They lack an internal model of their working environment and context, which often leads them to make irrational decisions. It's still safer to have a separate computer at home as a git server. 

Regarding the requirement to stop tasks being executed by Ailice, I have already implemented it in the latest version. Experiments show that using prompts for this purpose doesn’t work—on the bright side, this is a good thing, the error correction capability of the IACT architecture means that parent agents don’t trust the child agents when they claim tasks have been paused. I’ve added a hard interrupt feature in the code, and the "/stop" command in the UI to control it.
